tensorflow/python/training/monitored_session_test.py

```diff
@@ -1414,6 +1414,11 @@ def test_merge_caller_and_hook_run_options(self):
             isinstance(hook.run_metadata_list[0], config_pb2.RunMetadata))
         self.assertGreater(len(hook.run_metadata_list[0].partition_graphs), 0)
 
+  def test_with_statement_and_close(self):
+    # Test case for 12224
+    # No expection should be raised
+    with monitored_session.MonitoredSession() as session:
```

Comment:
context manager exit calls the close. sess.close should not be called by user if context manager is used.